src/migrations/indexed-event-params.js

```diff
@@ -0,0 +1,141 @@
+
+const getGraphCliVersion = sourceDir => {
+  let pkgJsonFile = path.join(
+    sourceDir,
+    'node_modules',
+    '@graphprotocol',
+    'graph-cli',
+    'package.json',
+  )
+  let data = fs.readFileSync(pkgJsonFile)
+  let jsonData = JSON.parse(data)
+  return jsonData.version
+}
+
+const dataSourceHasEventHandlers = dataSource =>
+  dataSource &&
+  typeof dataSource === 'object' &&
+  ((dataSource.mapping &&
+    typeof dataSource.mapping === 'object' &&
+    dataSource.mapping.eventHandlers &&
+    Array.isArray(dataSource.mapping.eventHandlers) &&
+    dataSource.mapping.eventHandlers.length > 0) ||
+    (dataSource.templates &&
+      Array.isArray(dataSource.templates) &&
+      dataSource.templates.reduce(
+        (hasEventHandlers, template) =>
+          hasEventHandlers || dataSourceHasEventHandlers(template),
+        false,
+      )))
+
+const maybeUpdateEventSignature = (manifestFile, dataSource, event) => {
+  let abiName = dataSource.getIn(['source', 'abi'])
+  let abis = dataSource.getIn(['mapping', 'abis'], immutable.List())
+  let abi = abis.find(abi => abi.get('name') === abiName)
+  let name = abi ? abi.get('name') : undefined
+  let file = abi ? abi.get('file') : undefined
+
+  let oldEventSignatures
+  let newEventSignatures
+  if (file && name) {
+    try {
+      abi = ABI.load(name, path.relative(path.dirname(manifestFile), file))
+    } catch (e) {
+      // Failed to load the ABI, return the event unchanged
+      return event
+    }
+
+    oldEventSignatures = abi.oldEventSignatures()
+    newEventSignatures = abi.eventSignatures()
+  }
+
+  if (oldEventSignatures && newEventSignatures) {
+    for (let i = 0; i < oldEventSignatures.size; i++) {
+      // Only replace the event signature if the signature of the event has changed
+      // and the old signature is not present in the new signatures anymore.
+      if (
+        event === oldEventSignatures.get(i) &&
+        oldEventSignatures.get(i) !== newEventSignatures[i] &&
+        !newEventSignatures.contains(event)
+      ) {
+        return newEventSignatures.get(i)
+      }
+    }
+  }
+
+  // Event signature doesn't need to be updated
+  return event
+}
+
+const maybeMigrateDataSource = (manifestFile, dataSource) => {
+  if (
+    dataSource.get('kind') === 'ethereum/contract' &&
+    immutable.List.isList(dataSource.getIn(['mapping', 'eventHandlers']))
+  ) {
+    dataSource = dataSource.updateIn(['mapping', 'eventHandlers'], eventHandlers =>
+      eventHandlers.map(eventHandler =>
+        eventHandler.update('event', event =>
+          maybeUpdateEventSignature(manifestFile, dataSource, event),
+        ),
+      ),
+    )
+  }
+
+  if (immutable.List.isList(dataSource.get('templates'))) {
+    dataSource = dataSource.update('templates', templates =>
+      templates.map(template => maybeMigrateDataSource(manifestFile, template)),
+    )
+  }
+
+  return dataSource
+}
+
+// If any of the manifest apiVersions are 0.0.1, replace them with 0.0.2
```

Comment:
This comments looks left over from a copy-paste.

Reply:
Indeed